memiavl/db.go

```diff
@@ -206,6 +212,31 @@ func Load(dir string, opts Options) (*DB, error) {
 	return db, nil
 }
 
+func removeTmpDirs(rootDir string, v int64) error {
+	entries, err := os.ReadDir(rootDir)
+	if err != nil {
+		return err
+	}
+
+	for _, entry := range entries {
+		if !entry.IsDir() {
+			continue
+		}
+		version, err := parseTmpVersion(entry.Name())
+		if err != nil {
+			continue
+		}
+		if version >= v {
```

Comment:
Why do pruning here, and it ignores the keep-recent settings

Reply:
I see the problem, we need to introduce readonly mode, only remove files in write mode.